proxymode/proxymode.go

```diff
@@ -0,0 +1,150 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package proxymode
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/elastic/package-registry/packages"
+	"github.com/elastic/package-registry/util"
+)
+
+type ProxyMode struct {
+	options ProxyOptions
+
+	httpClient     *http.Client
+	destinationURL *url.URL
+	resolver       *proxyResolver
+}
+
+type ProxyOptions struct {
+	Enabled bool
+	ProxyTo string
+}
+
+func NoProxy() *ProxyMode {
+	proxyMode, err := NewProxyMode(ProxyOptions{})
```

Comment:
Nit. Explicitly set enabled to false here, just in case `ProxyOptions` change.
```suggestion
	proxyMode, err := NewProxyMode(ProxyOptions{Enabled: false})
```

Reply:
Fixed